dependencies.yaml

```diff
@@ -382,8 +382,11 @@ dependencies:
     common:
       - output_types: [conda, requirements, pyproject]
         packages:
-          - dask>=2023.1.1
-          - distributed>=2023.1.1
+          - dask==2023.3.2
+          - distributed==2023.3.2.1
+      - output_types: conda
+        packages:
+          - dask-core==2023.3.2
```

Comment:
It looks like on the conda side,
- `dask-core` packages "just dask"
- `dask` packages dask + distributed (plus other tools? unclear if necessary)

Do we need to pin all of `dask`, `dask-core`, and `distributed`? Can we get away with pinning just `dask` or `dask-core` and `distributed` but not `dask`?

Some additional changes might be needed between pip and conda naming schemes -- that's okay, whatever we need to do is fine.

Reply:
> Can we get away with pinning just dask

Unfortunately not right now, as there are ongoing issues I've been working to resolve around the nightly versioning scheme that make it so it's possible to install stable `dask` but nightly `dask-core`/`distributed` 😕 

> or dask-core and distributed but not dask

This might be possible, though I would generally prefer keeping the `dask` metapackage around since it's essentially the only way to pull Dask's dataframe dependencies transitively, which I'm assuming we would want to stay in line with.

Reply:
Maybe we can add a comment that indicates something like `requirements: dask` is "actually dask" but `conda: dask` is a metapackage? I think it's worth being extra-clear what we mean in the `dependencies.yaml` file so that readers understand what's happening. I wouldn't have known that without reading the feedstock sources.

Reply:
Otherwise I'm on board with what you're saying @charlesbluca, thanks for the context.

Reply:
In a call, will add shortly

Reply:
Added the comment.